Source/MLXNN/Linear.swift

```diff
@@ -102,17 +102,21 @@ open class Linear: Module, UnaryLayer {
     ///   - inputDimensions: number of input dimensions
     ///   - outputDimensions: number of output dimensions
     ///   - bias: if `true` this layer will apply a bias
-    convenience init(inputDimensions: Int, outputDimensions: Int, bias: Bool = true) {
+    public convenience init(inputDimensions: Int, outputDimensions: Int, bias: Bool = true) {
         self.init(inputDimensions, outputDimensions, bias: bias)
     }
 
-    internal init(weight: MLXArray, bias: MLXArray? = nil) {
+    /// Initializer meant for subclasses to provide weight and bias arrays directly.
+    ///
+    /// This is used e.g. by ``QuantizedLinear`` to provide quantized weights and biases
+    /// rather than have ``Linear`` compute them.
+    public init(weight: MLXArray, bias: MLXArray? = nil) {
```

Comment:
Public allows subclasses outside the package to call this (sadly I don't see a way to get something like `protected` in Java which allows subclasses but not the general public to use).  Anyway, we need this so LoRALinear can call this and pass in the previous weight/bias, just like QuantizedLinear does.